app/operations/memberships/create.rb

```diff
@@ -12,7 +12,9 @@ def execute
       raise Unauthorized unless user_group.verify_join_token(join_token)
       raise Unauthorized unless user.id == api_user.id
 
-      Membership.create! user: api_user.user, user_group: user_group, state: :active
+      membership = Membership.find_or_initialize_by(user: api_user.user, user_group: user_group)
+      membership.state = :active
+      membership.tap(&:save!)
```

Comment:
what's the reason for tap here? won't it be new or existing model so save! will just work.

Reply:
#save returns true or false, the expected return value of this operation is the created object. Saved an extra "membership" line.

Reply:
:tada: